File: 0643. Maximum Average Subarray I.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function(nums, k) {
   let left = 0
   let right = 0
   let maxAve = -10000
   let n = nums.length
   let total = 0
   while(right < n) {
      if(right - left + 1  < k)  {
          total += nums[right]
          right ++
      } 
      else{
          total += nums[right];
          maxAve= Math.max(maxAve,total/k)
          total -= nums[left];
          left ++
          right ++
      }
   }
   return maxAve
};

File: 0643. Maximum Average Subarray I.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        left = right = avi = max_avi = 0
        n = len(nums)
        avi =  sum(nums[:k])
        max_avi = avi
        right = k
        while right < n:
            avi = avi + nums[right] - nums[left]
            max_avi = max(max_avi, avi)
            left, right = left+1, right+1
        return max_avi / k
